bpo-32529: imporved shutil.copyfileobj

YoSTEALTH opened the pull request:
Improved "copyfileobj" function to use less memory if provided Bytes IO and fixed potential encoding mismatch corruption in Text IO

<!-- issue-number: bpo-32529 -->
https://bugs.python.org/issue32529
<!-- /issue-number -->

YoSTEALTH committed:
Update shutil.py

YoSTEALTH committed:
Merge pull request #1 from YoSTEALTH/YoSTEALTH-patch-1

Update shutil.py

YoSTEALTH committed:
Update shutil.py

YoSTEALTH committed:
Update shutil.py

YoSTEALTH committed:
Update shutil.py

tiran reviewed (changes requested):
Before this patch gets merged, please provide memory and performance benchmarks for this change on the ticket

In case your benchmark show an improvement, you have to provide tests for the change, too.

YoSTEALTH committed:
Update shutil.py

YoSTEALTH committed:
Update shutil.py

After a good debate on #python i have removed the encoding checks as copyfileobj could potentially be used as encoding converter tool
